Ping msg

jwade109 opened the pull request:
Adds communication between drive_node and test_controller: drive_node regularly publishes Status messages to /robot/status, and must recieve an Int8 message atleast once every 2 seconds via /driver/ping or it will shut down until it does.

jwade109 committed:
Added rudimentary ping function

jwade109 committed:
Connection status can kill motors but isn't perfect yet

jwade109 committed:
Reworked robot-driver ping connection
drive_node now hibernates after 2 seconds without a ping from the driver station, and publishes a status message every second. Also split main.launch into robot.launch and driver.launch.

jwade109 committed:
Merge branch 'master' into ping_msg

jwade109 committed:
Fixed issues from merge

jwade109 committed:
Added respawn to launch files

jwade109 committed:
Merge remote-tracking branch 'origin' into ping_msg

jwade109 committed:
Switched ping message type for int8

jwade109 committed:
Just kidding now it's a Bool message

anidev committed:
Merge branch 'master' of github.com:VTAstrobotics/Robot-2017 into ping_msg

anidev commented on src/robot2017/src/drive_node.cpp:
```diff
@@ -41,31 +66,47 @@ int main(int argc, char **argv)
 
+    pub_status = nh.advertise<robot_msgs::Status>("/robot/status", 100);
+    driver_ping = nh.subscribe("/driver/ping", 1000, &recievedPing);
+    ros::Publisher fake_ping = nh.advertise<std_msgs::Bool>("/driver/ping", 100);
```
Doesn't look like this is necessary anymore

jwade109 replied:
correct, that was just for testing. removed

anidev commented on src/robot2017/src/drive_node.cpp:
```diff
@@ -41,31 +66,47 @@ int main(int argc, char **argv)
-    ros::Subscriber sub_tele = nh.subscribe("/robot/teleop", 1000, &RobotExec::teleopReceived, &exec);
-
-    ros::Subscriber sub_aut = nh.subscribe("/robot/autonomy", 1000, &RobotExec::autonomyReceived, &exec);
-
```
You also removed the subscriptions for teleop and autonomy - should add them back

jwade109 replied:
resolved. not sure why I removed them.

anidev commented on src/robot2017/src/drive_node.cpp:
```diff
@@ -41,31 +66,47 @@ int main(int argc, char **argv)
 
-        if (exec.isAutonomyActive())
+        double timeSince = (ros::Time::now() - lastDriverPing).toSec();
+        if (timeSince > 2 && !hibernating)
```
Could you replace the 2 with a constant int at the top of the file

anidev commented on src/robot2017/src/drive_node.cpp:
```diff
@@ -74,9 +115,8 @@ int main(int argc, char **argv)
             pub_fb.publish(motorFb);
         }
 
+        publishStatus();
```
It doesn't look you're actually setting the values for the status message anywhere before publishing

jwade109 replied:
not sure where you want it to be set. I figured there would be something in the ros::ok() loop.

anidev reviewed (changes requested):
Why did you change the launch files - they should probably be changed back, since each launch file is meant to be standalone, so the test launch needs to launch both drive_node as well as test. Also, the main launch file shouldn't have `-pc -debug -aut`, those are just for testing.

jwade109 committed:
Fixed drive_node, launch files
Changed the scope of a bunch of publishers/subscribers back. Readded test.launch, but keeping driver and robot.launch because they're useful for testing on two separate console windows.

Commit:
Add debouncing for mode toggle button (start button)

anidev committed:
Merge branch 'master' into debounce-modes

anidev committed:
Added folder for system install files

anidev committed:
Added package for sunrom (load cell library)

anidev committed:
Added comment for using set_Position in future

anidev committed:
Added gpio/adc library

anidev committed:
Removed unneeded todo

anidev committed:
Added sensors.cpp to robot2017 and fixed compile issues

anidev committed:
Implemented left/right load cells and added skeleton for status LEDs

anidev committed:
Added gpio for a kill button

anidev committed:
Implemented kill button and status LED in robot exec

anidev committed:
Added load cell weights to motor feedback message

Commit:
Add autonomy enable msg (std_msgs::Bool)

anidev committed:
Added functions for lift and storage potentiometers

anidev committed:
Sending pot data in publishMotors

Commit:
Updated robot_msgs for autonomy changes

anidev committed:
Updated to use changed messages and send remaining motor feedback data

anidev committed:
Changed lift to use active braking via set_Position

anidev committed:
Implemented rest of autonomyExec

anidev committed:
Tweaked debug message for robot kill

anidev committed:
Added uEnv.txt from beaglebone and updated corresponding README

anidev committed:
Renamed udev rules file to be correct

anidev committed:
Improved Sensors to not read data when testing

anidev committed:
Merge branch 'bbb_hw_libraries' into debounce-modes

anidev committed:
Fixed compile errors

anidev committed:
Changed publishMotors to publish data every time

anidev committed:
Changes to make sure onPC is respected for all motors and sensors

anidev committed:
Removed debug log for motor feedback

anidev committed:
Added debug print for autonomy toggle

anidev committed:
Merge branch 'bbb_hw_libraries' into ping_msg

anidev committed:
Commenting

anidev committed:
Removed digital estop code since no longer being used

anidev committed:
Merge branch 'bbb_hw_libraries' into ping_msg

anidev committed:
Consistent formatting

anidev committed:
readme

anidev committed:
Fixed segfault issue in gpio

anidev committed:
Removed storage pot and load cells

anidev committed:
Increased lift and storage speeds

anidev committed:
Added offset to pot angle

anidev committed:
Increased delay for GPIO export

anidev committed:
Added slow/fast teleop lift speed and custom auto lift speed

anidev committed:
Added soft limits for lift and storage

anidev committed:
Removed getting motor data output

anidev committed:
Fixed storage direction for autonomy

anidev committed:
Added battery voltage to motor feedback

anidev committed:
Should enforce limits when holding down button

anidev committed:
Fixed limit enforcing with continuous movement

anidev committed:
Merge branch 'bbb_hw_libraries' into ping_msg

anidev committed:
Removed unused variables

anidev committed:
Actually updates status before sending

anidev commented:
Finally been merged into branch bbb_hw_libraries
